### PR TITLE
WIP: weighted hamming distance in C

### PR DESCRIPTION
**Pull request by magnusja**

### CODE REVIEW

File: scipy/spatial/distance.py [hunk 1]
```diff
@@ -282,6 +282,15 @@ def _validate_minkowski_kwargs(X, m, n, **kwargs):
     return kwargs
 
 
+def _validate_whamming_kwargs(X, m, n, **kwargs):
+    if 'w' not in kwargs:
+        kwargs['w'] = 1.
+
+    w = kwargs['w']
```
Comment (tylerjereddy) [inline scipy/spatial/distance.py]: I think the few lines above can be simplified with something like `w = kwargs.get('w', 1.)`

**Comment by rgommers** [comment]
@magnusja if you have multiple little fixes/changes, it's best to all push them in one go rather than one by one. otherwise you're swamping our CI system with requests. and you can run tests locally first (e.g. `python -c "import scipy.spatial as s; s.test()"`).

**Comment by rgommers** [comment]
@magnusja accelerating the current implementation of `hamming` with `w` not None seems like a good idea, but there is no need for a new `whamming` metric. `hamming` itself already provides the API for weights.

**Comment by magnusja** [comment]
Hey @rgommers, thanks for the feedback. You mean more like this? https://github.com/magnusja/scipy/tree/feature/weighted-hamming

**Comment by rgommers** [comment]
Hi @magnusja, that looks about right from a quick browse. API wise definitely okay, will have to look at the C code later but looks reasonable. Can you push that branch to this PR?

**Comment by magnusja** [comment]
Unfortunately I cannot easily, so I am closing this in favor of: https://github.com/scipy/scipy/pull/9813

(some tests still fail unfortunately, I am still trying to figure out why)

**Pull request closed without merging**